tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -828,99 +816,94 @@ void EnqueueProgramCommand::assemble_device_commands(
-                            }
-                            uint32_t write_length, read_length;
-                            if (aligned_length <= max_length_per_sub_cmd) {
-                                read_length = aligned_length;
-                                write_length = read_length - padding;
-                            } else {
-                                read_length = max_paged_length_per_sub_cmd;
-                                write_length = read_length;
-                            }
-                            kernel_bins_dispatch_subcmds.back().emplace_back(CQDispatchWritePackedLargeSubCmd{
-                                .noc_xy_addr = noc_encoding,
-                                .addr = dst_addr,
-                                .length = (uint16_t)write_length,
-                                .num_mcast_dests = (uint16_t)dst_noc_info.second});
-                            dst_addr += write_length;
-
-                            kernel_bins_prefetch_subcmds.back().emplace_back(CQPrefetchRelayPagedPackedSubCmd{
-                                .start_page = (uint16_t)page_offset,
-                                .log_page_size = (uint16_t)HostMemDeviceCommand::LOG2_PROGRAM_PAGE_SIZE,
-                                .base_addr = base_address,
-                                .length = read_length});
-                            page_offset += read_length / HostMemDeviceCommand::PROGRAM_PAGE_SIZE;
-                            aligned_length -= read_length;
-                            kernel_bins_write_packed_large_data_aligned_sizeB.back() += read_length;
+                        noc_encoding = this->device->get_noc_unicast_encoding(this->noc_index, cores);
+                        write_linear = true;
+                    }
+                },
+                cores);
+            for (uint32_t kernel_idx = 0; kernel_idx < kg_transfer_info.dst_base_addrs.size(); kernel_idx++) {
+                if (write_linear) {
+                    kernel_bins_unicast_cmds.emplace_back(2 * CQ_PREFETCH_CMD_BARE_MIN_SIZE);
+                    cmd_sequence_sizeB += 2 * CQ_PREFETCH_CMD_BARE_MIN_SIZE;
+                    kernel_bins_unicast_cmds.back().add_dispatch_write_linear(
+                        false,            // flush_prefetch
+                        num_mcast_dests,  // num_mcast_dests
+                        noc_encoding,     // noc_xy_addr
+                        kg_transfer_info.dst_base_addrs[kernel_idx],
+                        kg_transfer_info.lengths[kernel_idx]);
+                    // Difference between prefetch total relayed pages and dispatch write linear
+                    uint32_t relayed_bytes =
+                        align(kg_transfer_info.lengths[kernel_idx], HostMemDeviceCommand::PROGRAM_PAGE_SIZE);
+                    // length_adjust needs to be aligned to NOC_DRAM_ALIGNMENT
+                    uint16_t length_adjust = uint16_t(relayed_bytes - kg_transfer_info.lengths[kernel_idx]);
```

Comment:
does this need to be explicitly aligned?

Reply:
No that was a legacy comment from when we were aligning it to dram alignment for some reason. I removed the alignment a while back but missed removing the comment. I'll remove.